app/src/main/java/com/example/henzoshimada/feeltrip/MainScreen.java

```diff
@@ -150,11 +163,43 @@ public void onCheckedChanged(CompoundButton arg0, boolean isChecked) {
             }
         });
 
+        ToggleButton toggleMostRecent = (ToggleButton) findViewById(R.id.toggleMostRecent);
+        toggleMostRecent.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
+
+            @Override
+            public void onCheckedChanged(CompoundButton arg0, boolean isChecked) {
+                //Toast.makeText(getApplicationContext(),"Toggled filter for friends!",Toast.LENGTH_SHORT).show();
+                FeelTripApplication.getFilterController().setMostrecentfilter(isChecked);
+                FeelTripApplication.loadFromElasticSearch();
+                FeelTripApplication.getMoodAdapter(getBaseContext()).notifyDataSetChanged();
+            }
+        });
+
+        ImageButton searchButton = (ImageButton) findViewById(R.id.imageButton);
+        searchButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                searchKeyword(v);
+            }
+        });
+
+
     }
 
     @Override
-    protected void onStart() {
-        super.onStart();
+    protected void onResume() {
+        super.onResume();
+
+        ToggleButton toggleFriends = (ToggleButton) findViewById(R.id.toggleFriends);
+        TextView textView = (TextView) findViewById(R.id.textView5);
+        if(FeelTripApplication.getFrag().equals("profile")) {
+            toggleFriends.setVisibility(View.GONE);
+            textView.setVisibility(View.GONE);
+        } else {
```

Comment:
Hide the "Friends Only" filter option while we're on the Profile fragment, since it's useless on that screen anyways.